### ISSUE TITLE
Add to_radiance method to energy construction and materials

### ISSUE THREAD

**Issue by mostaphaRoudsari** [ladybug-tools/honeybee-energy#9]
See here: https://github.com/ladybug-tools/honeybee-energy/pull/8#issuecomment-503770502

@mostaphaRoudsari , Two questions that popped into my mind about how you imagine the to_radiance methods working:

    Should the method simply return a Radiance string or should it return a honeybee_radiance material object (in which case , it should be be code-injected from honeybee_radiance)?
    It would be really nice to use genbsdf to generate a BSDF material of venetian blinds in the case that a blind material is in the window construction. Do you have any recommendations for how to implement this (ie. should it be written into the xml file and the file path put into the Radiance string)? How are you currently handing external BSDF files in honeybee_radiance?

**Comment by mostaphaRoudsari** [comment]
Pinging @rpg777! Hey Rob, is there a reference, rather than the source code of daylight measure, that you can point us to for understand how OpenStudio handles different cases of windows with dynamic blinds? I know that you have been back and forth between different solutions. Thanks! :)

**Comment by rpg777** [comment]
Yeah so the way it's handled in the Radiance daylight measure is that it's not really handled. It's completely divorced from the E+ simulation, and you only have the option of choosing a ven blind or a shade, the transmittances (and in the case of the blinds, the slat configuration) are fixed, and there is a small selection of BSDFs that ship with OpenStudio, which get mapped to the user's choice in the... plugin. That's right, these shade controls were implemented as the old User Scripts and were never fully integrated into the App gui. Pretty standard support and follow-through for all of this, which is to say, there was none. 

The specifics of the handling are in the Radiance forward translator if I recall correctly. There's a bit in there that translates the shade controls to mappings in the `mapping.rad` file, the whole mess starts here: https://github.com/NREL/OpenStudio/blob/develop/openstudiocore/src/radiance/ForwardTranslator.cpp#L1596

**Comment by chriswmackey** [comment]
Thank you for the insight, @rpg777 .  I see that it's difficult to translate the blind materials from E+ to Radiance and having a collection of pre-exported BSDF files is one way to deal with it.  For now, I will just raise a `NotImplementedError` if someone translates an E+ construction with blind materials and hopefully, at some point in the future, we can attempt to implement your ideal scenario of "support BSDFs in a bidirectional way".   That is quite an awesome pun, BTW.  Would you mind if we borrowed it if we ever end up successfully implementing it and need a title for a workshop?

Lastly, when you get the chance, would you mind sharing your thoughts on [these 30 lines of code that we have written for the `to_radiance()` method in our honeybee_energy construction object](https://github.com/ladybug-tools/honeybee-energy/blob/master/honeybee_energy/construction.py#L933-L964)
I would think that using a Radiance glass material with a transmittance of the EP construction would be a suitable for most clear windows but I was less sure of whether it is good practice to use a Trans material for light diffusing layers like the E+ shade material. I know that E+ models these shades as perfect diffusers so, if we are trying to match this, is it correct to make a Trans material with 0 beam transmittance and a diffuse transmittance equal to the shade transmittance?

**Comment by rpg777** [comment]
Haha, no problem, just provide a credit on the title slide. ;) 

I'll take a look at that code ASAP...

**Comment by rpg777** [comment]
OK so yeah ASAP turned out to be "immediately". Two things:
1. You need to convert transmittance to transmissivity and use that for the `glass` primitive in Radiance. 
2. Yeah, if parity between the modeling tools (and their primitives) is the main goal--which it should be in this case--then yes, on the Radiance side, use `trans` with zero specular transmittance. Here again tho, make sure you are calculating these values correctly. It's pretty straightforward for grey, but if there are non-equal RGB values, you need to follow it all through the formula(e). P.S. this is what we do in OpenStudio, too:
https://github.com/NREL/OpenStudio/blob/develop/openstudiocore/src/radiance/ForwardTranslator.cpp#L1306-L1357

**Comment by mostaphaRoudsari** [comment]
Thank you @rpg777 for the immediate reply!

> 1. You need to convert transmittance to transmissivity and use that for the glass primitive in Radiance.

That is taken care of under the hood in `honeybee-radiance`. Line 197 here:

https://github.com/ladybug-tools/honeybee-radiance/blob/093eed49719dbef832b7f58b80c09e09ad4c9c78/honeybee_radiance/primitive/material/glass.py#L176-L202

> 2. ...make sure you are calculating these values correctly. It's pretty straightforward for grey, but if there are non-equal RGB values, you need to follow it all through the formula(e).

This is also taken care of in `honeybee-radiance`:

https://github.com/ladybug-tools/honeybee-radiance/blob/093eed49719dbef832b7f58b80c09e09ad4c9c78/honeybee_radiance/primitive/material/trans.py#L152-L221

But we need to use `from_reflected_specularity` instead of `from_single_reflectance` for non-gray cases. Thank you for catching this.

**Comment by chriswmackey** [comment]
Thank you @rpg777 .  For now, I think we will stick to just supporting grey translators in the `to_radiance` methods since the E+ materials don't have color information and the two main use cases where these methods will be called don't require colors:
1. Calculating illuminance and annual daylight from energy models.
2. Calculating total shortwave reflected solar irradiance to estimate mean radiant temperature for thermal comfort studies.
It is good to know what's involved in supporting color information, though, and this will likely come in handy down the line.

**Comment by chriswmackey** [comment]
I just wanted to link to an issue on honeybee-radiance where we came up with a plan for this:
https://github.com/ladybug-tools/honeybee-radiance/issues/35#issuecomment-615951173

I think we will ultimately make two pathways to modeling dynamic windows in honeybee-energy:
1. A sophisticated and highly flexible workflow using the EMS and the strategy described in that issue.
2. A simple DynamicWindowConstruction will all of the limitations of EnergyPlus ShadingControl but is super-easy to set up.
